### PR TITLE
[Java] Enable concurrent calls in local mode.

### PR DESCRIPTION
**Pull request by jovany-wang**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: java/runtime/src/main/java/io/ray/runtime/task/LocalModeTaskSubmitter.java [hunk 1]
```diff
@@ -460,4 +472,11 @@ private static JavaFunctionDescriptor getJavaFunctionDescriptor(TaskSpec taskSpe
     }
     return returnIds;
   }
+
+  /** Whether this is an concurrent actor. */
```
Comment (kfstorm) [inline java/runtime/src/main/java/io/ray/runtime/task/LocalModeTaskSubmitter.java:476]:
```suggestion
  /** Whether this is a concurrent actor. */
```

Comment (kfstorm) [reply java/runtime/src/main/java/io/ray/runtime/task/LocalModeTaskSubmitter.java:476]: What's the relationship between the task spec and the concurrent actor?

Comment (jovany-wang) [reply java/runtime/src/main/java/io/ray/runtime/task/LocalModeTaskSubmitter.java:476]: Just fetch the maxConcurrency field from actor_creation_task_spec. None if it's not a creation spec. (PS: Similar with the method `getActorId(spec)`)

Comment (kfstorm) [reply java/runtime/src/main/java/io/ray/runtime/task/LocalModeTaskSubmitter.java:476]: Can you update the comment in code?

**Comment by kfstorm** [comment]
Buildkite test failed.

**Review by raulchen — APPROVED** [review]

**Comment by jovany-wang** [comment]
Buildkite should be fixed at this commit https://github.com/ray-project/ray/pull/14896/commits/6c94760c379c4847895e92c6c6620adb1be75da6

**Pull request merged**